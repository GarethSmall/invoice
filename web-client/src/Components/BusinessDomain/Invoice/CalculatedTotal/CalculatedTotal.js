// @flow
import React, { PureComponent } from 'react';
import { moneyFormat } from '../../../../Helpers/Math/math';
import { OrderItem } from 'shared-flow';

type Props = {
  items : OrderItem[],
  taxPercent : number,
};

export default class CalculatedTotal extends PureComponent<Props> {
  static defaultProps = {
    taxPercent: 5,
  };
  render() {
    const { items, taxPercent } = this.props;
    const priceReducer = (accumulator : number, current : OrderItem) => accumulator + current.price;
    const subTotal : number = items.reduce(priceReducer, 0);
    const tax : number = subTotal * (taxPercent / 100);
    const total : number = subTotal + tax;
    return (
      <div>
        <div>Subtotal: {moneyFormat(subTotal)}</div>
        <div>Tax (5%): {moneyFormat(tax)}</div>
        <div>Total: {moneyFormat(total)}</div>
      </div>
    )
  }
}