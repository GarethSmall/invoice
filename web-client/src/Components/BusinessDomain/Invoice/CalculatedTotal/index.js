//@flow
import CalculatedTotal from './CalculatedTotal';
export default CalculatedTotal;
