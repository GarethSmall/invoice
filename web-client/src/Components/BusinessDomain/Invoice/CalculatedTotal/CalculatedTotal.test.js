// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import shallow from 'enzyme/shallow';
import CalculatedTotal from './';
const { describe, test, expect } = global;

Enzyme.configure({ adapter: new Adapter() });

describe('CalculatedTotal', () => {
  test('should render', () => {

    const mockItems = [
      {
        _id: '1',
        order: '1',
        product: {
          _id: '1',
          name: 'Test T-shirt',
          price: 32.00,
        },
        quantity: 2,
        price: 64.00,
      },
      {
        _id: '2',
        order: '1',
        product: {
          _id: '2',
          name: 'Test Jeans',
          price: 75.00,
        },
        quantity: 2,
        price: 150.00,
      },
    ];
    const component = shallow((
      <CalculatedTotal items={mockItems} />
    ));
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
