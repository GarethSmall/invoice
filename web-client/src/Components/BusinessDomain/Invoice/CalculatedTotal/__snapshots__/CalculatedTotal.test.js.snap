// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatedTotal should render 1`] = `
<div>
  <div>
    Subtotal: 
    $214.00
  </div>
  <div>
    Tax (5%): 
    $10.70
  </div>
  <div>
    Total: 
    $224.70
  </div>
</div>
`;
